Stratis.Bitcoin/Features/BlockStore/LoopSteps/BlockStoreInnerStepReadBlocks.cs

```diff
@@ -41,65 +42,69 @@ public BlockStoreInnerStepReadBlocks(ILoggerFactory loggerFactory)
         public override async Task<InnerStepResult> ExecuteAsync(BlockStoreInnerStepContext context)
         {
             this.logger.LogTrace("()");
-
-            if (!context.DownloadStack.Any())
+			if (!context.DownloadStack.Any())
             {
                 this.logger.LogTrace("(-):{0}", InnerStepResult.Stop);
                 return InnerStepResult.Stop;
             }
+            
 
-            BlockPuller.DownloadedBlock downloadedBlock;
-
-            ChainedBlock nextBlock = context.DownloadStack.Peek();
+            var loopCount = BlockStoreInnerStepContext.DownloadStackThreshold;
 
-            if (context.BlockStoreLoop.BlockPuller.TryGetBlock(nextBlock, out downloadedBlock))
+            while (--loopCount > 0)
```

Comment:
I'm not sure that this is the right thing to do as by the time we get here the outer loop would only be running this InnerStep and as such the loop seems a bit unnecessary? Basically, we'll continue to call TryGetBlock until the DownloadStack is empty, then we'll stop and the outer BS loop will start again.